bpf/bpf_lxc.c

```diff
@@ -2166,17 +2162,6 @@ int tail_ipv4_to_endpoint(struct __ctx_buff *ctx)
 		proxy_redirect = true;
 		break;
 	case CTX_ACT_OK:
-#if !defined(ENABLE_ROUTING) && defined(TUNNEL_MODE) && !defined(ENABLE_NODEPORT)
-		/* In tunneling mode, we execute this code to send the packet from
-		 * cilium_vxlan to lxc*. If we're using kube-proxy, we don't want to use
-		 * redirect() because that would bypass conntrack and the reverse DNAT.
-		 * Thus, we send packets to the stack, but since they have the wrong
-		 * Ethernet addresses, we need to mark them as PACKET_HOST or the kernel
-		 * will drop them.
-		 * See #14646 for details.
-		 */
-		ctx_change_type(ctx, PACKET_HOST);
-#endif /* !ENABLE_ROUTING && TUNNEL_MODE && !ENABLE_NODEPORT */
```

Comment:
That removal looks correct to me. This code was correctly added in one place per IP family in https://github.com/cilium/cilium/pull/14675/files, but then someone (:wink: :wink:) [duplicated](https://github.com/cilium/cilium/commit/0f11ce8d87c2fd30b5765fb144c97711e6b287a2) this code and added it here as well, where it's not necessary.

Might be worse explaining in the commit message how we arrived in this place as otherwise the code removal looks very worrying. Maybe not worth repushing though.

Reply:
> That removal looks correct to me. This code was correctly added in one place per IP family in https://github.com/cilium/cilium/pull/14675/files, but then someone (😉 😉) [duplicated](https://github.com/cilium/cilium/commit/0f11ce8d87c2fd30b5765fb144c97711e6b287a2) this code and added it here as well, where it's not necessary.
> 

Thank you! I'm actually happy with how that refactor played out :). Start by blindly lifting a piece of code from a common path into the callers, then realize that one of the callers never even needed it.

> Might be worse explaining in the commit message how we arrived in this place as otherwise the code removal looks very worrying. Maybe not worth repushing though.

ack, will do that if a repush is needed :+1: